### PR TITLE
Unify use of yield/yield_k

### PR DESCRIPTION
**Pull request by hkaiser**
- cleanup configuration of TLS and compiler specific fence operation

### CODE REVIEW

File: hpx/config/compiler_native_tls.hpp [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+#if defined(__has_feature)
+#  if __has_feature(cxx_thread_local)
+#    define HPX_NATIVE_TLS thread_local
+#  endif
+#elif defined(HPX_HAVE_CXX11_THREAD_LOCAL)
+#  define HPX_NATIVE_TLS thread_local
+#endif
+
+#if !defined(HPX_NATIVE_TLS)
+#  if defined(_GLIBCXX_HAVE_TLS)
```
Comment (sithhell) [inline hpx/config/compiler_native_tls.hpp]: There needs to be at least one C++ header include to have this define, maybe `#include <cstdint>`?

Comment (sithhell) [reply hpx/config/compiler_native_tls.hpp]: Or even `<ciso646>`...

Comment (hkaiser) [reply hpx/config/compiler_native_tls.hpp]: Fair point, will fix. Thanks!

Comment (hkaiser) [reply hpx/config/compiler_native_tls.hpp]: Done.

**Pull request merged**